addons/web/static/src/core/user_service.js

```diff
@@ -37,6 +38,20 @@ export const userService = {
                 }
                 return groupProms[group];
             },
+            async checkAccessRight(model, operation) {
+                const key = model + operation;
+                if (!(key in cache)) {
+                    const url = `/web/dataset/call_kw/${model}/check_access_rights`;
+                    cache[key] = await rpc(url, {
```

Comment:
you should store promises in the cache, not values, because if you call this method twice consecutively, you'll do the rpc twice (check hasGroup).
Also, maybe rename into accessRightCache?

Reply:
yep. i finally decided to add a cache utility class.